### PR TITLE
 201-key-vault-secret-create | propsoing secureObject wrapper for array 

### PR DESCRIPTION
**Pull request by danielstocker**
adding secureObject wrapper to align more with ARM templating recommendations: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-authoring-templates#parameters

### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

[x] Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* changed variable input to secureObject rather than array
* added object wrapper around array
* changed reference expressions in template to support wrapper
* using a secureObject prevents other users from seeing the secrets in the portal after the deployment

### Screenshots of the difference

Before: (everyone can see the secrets)
secret-value-before-wrapper

After: (secrets hidden)
using-secret-wrapper

### CODE REVIEW

File: 201-key-vault-secret-create/azuredeploy.json [hunk 1]
```diff
@@ -54,10 +54,10 @@
             }
         },
         "secrets": {
-            "type": "array",
+            "type": "secureObject",
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.json]: I like this but it took me a while to wrap my head around what you're doing given the naming convention...  Can we use "secretObject" for the parameter name and "secrets" for the objectName?

Comment (danielstocker) [inline 201-key-vault-secret-create/azuredeploy.json]: Hi Brian, yeah I agree it makes more sense if we call the wrapper "secureObject" and the array in it "secrets". I have made these adjustments now.

File: 201-key-vault-secret-create/azuredeploy.json [hunk 2]
```diff
@@ -53,11 +53,11 @@
                 "description": "Specifies if the vault is enabled for volume encryption"
             }
         },
-        "secrets": {
-            "type": "array",
+        "secureObject": {
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.json]: let's just call this "secretsObject" to avoid confusion with the typeName...

Comment (danielstocker) [inline 201-key-vault-secret-create/azuredeploy.json]: okay yeah that's a fair point

**Pull request merged**